File: ROID-Forum-Web-Client/Convert Client Code/GMGWebsite/web/Website/Sections/AllSection/ThreadViews/HeaderView.dart
library AllThreadHeaderView;
import 'dart:html';
import '../ThreadInfo.dart';
import '../ThreadController.dart';

class HeaderView {
  DivElement headerDiv;
  SpanElement threadHeaderTitle, threadHeaderOwner, threadHeaderSection, threadHeaderComment;
  ImageElement threadHeaderImage;
  ThreadInfo thread;
  ThreadController threadController;

  /**********Make GUI Components***********/
  HeaderView(ThreadInfo t, ThreadController tc) {
    thread = t;
    threadController = tc;
    makeThreadHeaderDivComponent();
    makeThreadHeaderTitleComponent();
    makeThreadHeaderOwnerComponent();
    makeThreadHeaderSectionComponent();
    makeThreadHeaderCommentComponent();
    makeThreadHeaderImageComponent();
  }
  DivElement getDiv() {
    return headerDiv;
  }
  void onClick() {
    threadController.threadClicked(thread);
  }
  void updateTitle() {
    threadHeaderTitle.text = thread.getTitle();
  }
  void updateOwner() {
    threadHeaderOwner.text = thread.getOwner();
  }
  void updateAvatarURL() {
    threadHeaderImage.src = thread.getAvatarURL();
  }
  void updateCommentCount() {
    threadHeaderComment.text = "Comments: ${thread.getCommentCount()}";
  }
  void updateSection() {
    threadHeaderSection.text = thread.getSection();
  }
  void makeThreadHeaderDivComponent() {
    headerDiv = new DivElement();
    headerDiv.style
        ..position = "absolute"
        ..width = "calc(100% - 20px)"
        ..top = "0px"
        ..left = "10px"
        ..height = "80px"
        ..borderTopStyle = "solid"
        ..borderWidth = "1px"
        ..borderColor = "#AAAAAA";
  }
  void makeThreadHeaderTitleComponent() {
    threadHeaderTitle = new SpanElement();
    threadHeaderTitle.style
        ..position = "absolute"
        ..fontSize = "30px"
        ..left = "70px"
        ..top = "10px"
        ..textDecoration = "underline"
        ..textOverflow = "ellipsis"
        ..maxWidth = "calc(100% - 70px)"
        ..height = "30px"
        ..cursor = "pointer"
        ..whiteSpace = "nowrap"
        ..overflow = "hidden"
        ..paddingTop = "10px";
    headerDiv.append(threadHeaderTitle);
    threadHeaderTitle.onClick.listen((var d) {
      onClick();
    });
  }
  void makeThreadHeaderOwnerComponent() {
    threadHeaderOwner = new SpanElement();
    threadHeaderOwner.style
        ..position = "absolute"
        ..fontSize = "15px"
        ..right = "10px"
        ..bottom = "10px"
        ..cursor = "pointer";
    headerDiv.append(threadHeaderOwner);
  }
  void makeThreadHeaderSectionComponent() {
      threadHeaderSection = new SpanElement();
      threadHeaderSection.style
          ..position = "absolute"
          ..width = "140px"
          ..fontSize = "15px"
          ..right = "calc(50% - 70px)"
          ..textAlign = "center"
          ..bottom = "10px"
          ..cursor = "pointer";
      headerDiv.append(threadHeaderSection);
    }
  void makeThreadHeaderCommentComponent() {
    threadHeaderComment = new SpanElement();
    threadHeaderComment.style
        ..position = "absolute"
        ..fontSize = "15px"
        ..left = "70px"
        ..bottom = "10px"
        ..cursor = "pointer";
    headerDiv.append(threadHeaderComment);
    threadHeaderComment.onClick.listen((var d) {
      onClick();
    });
  }
  void makeThreadHeaderImageComponent() {
    threadHeaderImage = new ImageElement();
    threadHeaderImage.style
        ..position = "absolute"
        ..left = "0px"
        ..top = "10px"
        ..height = "60px"
        ..width = "60px"
        ..backgroundColor = "rgb(200,200,200)"
        ..borderStyle = "solid"
        ..borderWidth = "1px";
    headerDiv.append(threadHeaderImage);
  }
}

File: ROID-Forum-Web-Client/src/Interface/Section/ThreadViews/HeaderView.ts
import './HeaderView.scss';

import {ThreadInfo} from "../ThreadInfo";
import {ThreadController} from "../ThreadController";
import {Interface} from "../../../Utility/Interface";

export class HeaderView {
    headerDiv: HTMLDivElement;
    threadHeaderTitle: HTMLSpanElement;
    threadHeaderOwner: HTMLSpanElement;
    threadHeaderComment: HTMLSpanElement;
    threadHeaderImage: HTMLImageElement;
    thread: ThreadInfo;
    threadController: ThreadController;

    constructor(t: ThreadInfo, tc: ThreadController, darkTheme : boolean) {
        this.thread = t;
        this.threadController = tc;

        this.headerDiv = Interface.Create({type: 'div', className: 'HeaderView', onClick: this.onClick, elements: [
            this.threadHeaderTitle = Interface.Create({type: 'span', className: 'HeaderTitle'}),
            this.threadHeaderOwner = Interface.Create({type: 'span', className: 'HeaderOwner'}),
            this.threadHeaderComment = Interface.Create({type: 'span', className: 'HeaderCommentCount'}),
            this.threadHeaderImage = Interface.Create({type: 'img', className: 'HeaderOwnerAvatarImage'})
        ]});
        if (darkTheme) {
            this.headerDiv.classList.add('DarkTheme');
        } else {
            this.headerDiv.classList.add('LightTheme');
        }
    }

    getDiv(): HTMLDivElement {
        return this.headerDiv;
    }

    onClick = () => {
        this.threadController.threadClicked(this.thread);
    };

    updateTitle() {
        this.threadHeaderTitle.innerText = this.thread.getTitle();
    }

    updateOwner() {
        this.threadHeaderOwner.innerText = this.thread.getCreatorDisplayName();
    }

    updateAvatarURL() {
        if (this.thread.getAvatarURL() == null || this.thread.getAvatarURL() == '' || this.thread.getAvatarURL() == undefined) {
            return;
        }
        this.threadHeaderImage.src = this.thread.getAvatarURL();
    }

    updateCommentCount() {
        this.threadHeaderComment.innerText = `Comments: ${this.thread.getCommentCount()}`;
    }
}